lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -232,18 +245,25 @@ private static function parseDatabaseUrl(array $params)
         // (pdo_)?sqlite3?:///... => (pdo_)?sqlite3?://localhost/... or else the URL will be invalid
         $url = preg_replace('#^((?:pdo_)?sqlite3?):///#', '$1://localhost/', $params['url']);
 
-        $url = parse_url($url);
+        // PHP < 5.4.8 doesn't parse schemeless urls properly.
+        // See: https://php.net/parse-url#refsect1-function.parse-url-changelog
+        if (PHP_VERSION_ID < 50408 && strpos($url, '//') === 0) {
+            $url = parse_url('fake:' . $url);
+
+            unset($url['scheme']);
+        } else {
+            $url = parse_url($url);
+        }
 
         if ($url === false) {
             throw new DBALException('Malformed parameter "url".');
         }
 
-        if (isset($url['scheme'])) {
-            $params['driver'] = str_replace('-', '_', $url['scheme']); // URL schemes must not contain underscores, but dashes are ok
-            if (isset(self::$driverSchemeAliases[$params['driver']])) {
-                $params['driver'] = self::$driverSchemeAliases[$params['driver']]; // use alias like "postgres", else we just let checkParams decide later if the driver exists (for literal "pdo-pgsql" etc)
-            }
-        }
+        // If we have a connection URL, we have to unset the default PDO instance connection parameter (if any)
+        // as we cannot merge connection details from the URL into the PDO instance (URL takes precedence).
+        unset($params['pdo']);
```

Comment:
Yeah so since I originally wrote this URL stuff... the idea is definitely that a URL can override any individual defaults that are set. The reason is that were it the other way round, you'd have to repeat all individual args to unset the URL instructions. Also, the idea is kind of to have a few defaults for e.g. all dev envs set individually through a framework or whatever, and then a URL can "fill" or "override" those. Think charset, server version and SSL mode set "globally" through individual parameters, and then each developer/environment just has a single `mysql://user:pass@host:port/dbname` URL parameter set, but they could always attach `?sslmode=none` to that URL for testing to override the default.